README.md

```diff
@@ -27,34 +27,25 @@ The Python wrapper automatically compiles the C++ library (cirronlib.cpp) on fir
 ```
 from cirron import Collector
 
-collector = Collector()
-
 # Start collecting performance metrics
-collector.start()
-
-# Your code here
-# ...
+with Collector() as collector:
+    # Your code here
+    # ...
 
-# Stop collecting and retrieve the metrics
-metrics = collector.end()
-print(metrics)
+# Retrieve the metrics
+print(collector.counters)
```

Comment:
The print can be moved inside the context (at least I'd call it there as I wouldn't be sure if counters/other vars/state would be available after the collector ends/closes). Same for tracer, unless it's intentional to leave that to demonstrate that the variable still exists after the context is closed or for some other reason.

Reply:
`counters` is only filled in `__exit__`, so it's not valid within the `with` scope.